src/coprocessor/codec/mysql/duration.rs

```diff
@@ -7,16 +7,26 @@ use std::time::Duration as StdDuration;
 use std::{i64, str, u64};
 use tikv_util::codec::number::{self, NumberEncoder};
 use tikv_util::codec::BytesSlice;
-use time::{self, Tm};
 
-use super::super::Result;
-use super::{check_fsp, parse_frac, Decimal};
+use nom::character::complete::{digit1, multispace0, multispace1};
+use nom::{
+    alt_complete, call, char, complete, cond_with_error, do_parse, eof, map, map_res, opt, peek,
+    preceded, tag, IResult,
+};
+
+use super::super::{Result, TEN_POW};
+use super::{check_fsp, Decimal};
 
 pub const NANOS_PER_SEC: i64 = 1_000_000_000;
 pub const NANO_WIDTH: u32 = 9;
+
 const SECS_PER_HOUR: u64 = 3600;
 const SECS_PER_MINUTE: u64 = 60;
 
+const MAX_HOURS: u32 = 838;
+const MAX_MINUTES: u32 = 59;
+const MAX_SECONDS: u32 = 59;
+
 /// `MAX_TIME_IN_SECS` is the maximum for mysql time type.
 const MAX_TIME_IN_SECS: u64 = 838 * SECS_PER_HOUR + 59 * SECS_PER_MINUTE + 59;
```

Comment:
Replace the number literal to const is better because we have defined them.

Reply:
The `MAX_TIME_IN_SECS` will be useless in my next PR because I change the representation of `Duration`. But I will fix it though.